### PR TITLE
Use and validate the configs required for `view`

### PR DESCRIPTION
**Pull request by winhamwr**

### CODE REVIEW

File: neckbeard/actions/up.py [hunk 1]
```diff
@@ -116,87 +101,98 @@ def up(force='n'):
...
+        )
+        if seed_deployment_name:
+            seed_config = configuration_manager.get_environment_config(
+                seed_deployment_name,
+            )
+            seed_deployment = Deployment(
+                seed_deployment_name,
+                seed_config.get('ec2', {}),
+                seed_config.get('rds', {}),
+                seed_config.get('elb', {}),
```
Comment (jlward) [inline neckbeard/actions/up.py:137]: Have you thought about making all these config guys defaultdicts?

File: neckbeard/actions/up.py [hunk 2]
```diff
@@ -116,87 +101,98 @@ def up(force='n'):
...
                 )
-                rds_deployers.append(deployer)
+
+                if aws_type == 'ec2':
+                    ec2_deployers.append(deployer)
+                elif aws_type == 'rds':
+                    rds_deployers.append(deployer)
+
+    # We don't actually want to do deployments until we have tests
+    assert False
```
Comment (jlward) [inline neckbeard/actions/up.py:195]: Just want to make sure this is still the case.

File: neckbeard/bin/neckbeard.py [hunk 3]
```diff
@@ -100,6 +101,18 @@ def run_commands(command, environment, configuration_directory):
...
         return 1
 
+    if environment is None:
+        # If no environment is given, but there's only one environment
+        # available, just go ahead and use it
+        available_environments = configuration.get_available_environments()
+        if len(available_environments) == 1:
+            environment = available_environments[0]
+        else:
+            raise Exception((
```
Comment (jlward) [inline neckbeard/bin/neckbeard.py]: Shouldn't you be using a more specific exception that `Exception`?

File: neckbeard/cloud_resource.py [hunk 4]
```diff
@@ -67,6 +80,10 @@ def __str__(self):
 
         return super(InfrastructureNode, self).__str__()
 
+    def save(self):
+        logger.critical("Called save on %s", self)
+        return
```
Comment (jlward) [inline neckbeard/cloud_resource.py:88]: Is this supposed to be doing nothing (except logging)?

**Comment by jlward** [comment]
Looks good, after my comments.

File: neckbeard/actions/up.py (hunk 1 above)
Comment (winhamwr) [inline neckbeard/actions/up.py:137]: This method of passing in the configs by node type is going to go away entirely. We'll pass in a dictionary of "cloud resource" configs. If we weren't, though, defaultdict would remove some boiler plate, agreed.

File: neckbeard/actions/up.py (hunk 2 above)
Comment (winhamwr) [inline neckbeard/actions/up.py:195]: Yup. I committed this and the override on `neckbeard.cloud_resource:InfrastructureNode.save` to stop actual changes from happening. I don't want to break the beta deploy until I've thought through all of the code and (hopefully) added [moto](https://github.com/spulec/moto)-based mocked-out tests for everything. For the next neckbeard ticket, this `assert False` will probably move to both the `neckbeard.cloud_provisioners` for ec2 and rds.

File: neckbeard/bin/neckbeard.py (hunk 3 above)
Comment (winhamwr) [inline neckbeard/bin/neckbeard.py]: I really should. This was just kind of lazy and not actually consistent with the return code error strategy the other things were using.

File: neckbeard/cloud_resource.py (hunk 4 above)
Comment (winhamwr) [inline neckbeard/cloud_resource.py:88]: Added a comment explaining what's up.

**Pull request merged**